File: backend/src/models/comment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Comment struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	PostID    uint           `json:"post_id" gorm:"not null;index"`
	UserID    uint           `json:"user_id" gorm:"not null;index"`
	ParentID  *uint          `json:"parent_id" gorm:"index"`
	Body      string         `json:"body" gorm:"not null;size:500"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"deleted_at" gorm:"index"`

	Author User `json:"author" gorm:"foreignKey:UserID"`
}

File: frontend/src/types/comment.ts
export interface CommentNode {
  id: string;
  post_id: string;
  user_id: string;
  parent_id?: string | null;
  body: string;
  created_at: Date;
  updated_at: Date;
  author_username?: string;
  author_display_name?: string;
  children?: CommentNode[];
}

export interface CreateCommentData {
  body: string;
  parent_id?: string | null;
}
